### PR TITLE
feat: :fire: added tobot_tooling

### PR DESCRIPTION
**Pull request by FrederikTobner**
added tobot_tooling as a library, moved vscode config to the technical documentation and improved cmake scripts

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -1,17 +0,0 @@
-{
```
Comment (jksevend) [inline .vscode/launch.json:1]: Shouldnt we keep debugging?

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,16 +2,18 @@ cmake_minimum_required(VERSION 3.18.0)
...
 
 add_compile_definitions(LOGS_FOLDER_PATH="${PROJECT_SOURCE_DIR}/logs/")
 
-add_subdirectory(src)
-
-# googletest tests
+# enabling tests
 enable_testing()
-add_subdirectory(tests)
+# building tooiling project
```
Comment (jksevend) [inline CMakeLists.txt:13]: Please add a `packages` subdirectory where tobot_engine and tobot_tooling will live.

File: example/example.cpp [hunk 3]
```diff
@@ -3,7 +3,7 @@
 class Player : public Tobot::Core::GameEntity {
     public:
         Player() : GameEntity("player", 50, 50) {
-            this->setTexture(Tobot::Core::AssetManager::loadTexture("../assets/inspirational_picture.png"));
+            this->setTexture(Tobot::Core::AssetManager::loadTexture("../../assets/inspirational_picture.png"));
```
Comment (jksevend) [inline example/example.cpp:6]: Lets hope this works, we should really take this one up next

File: tobot_tooling/src/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+set(TOBOT_TOOLING_SOURCE_FILES
+"logger.cpp"
```
Comment (jksevend) [inline tobot_tooling/src/CMakeLists.txt:2]: Is tobot_asserts included somewhere else when debug config is selected?

**Comment by jksevend** [comment]
Please also rename `tobot.h` to `tobot_engine.h`. 
Is it possible to create a separate include directory for `tobot_tooling` where all headers are exported through a `tobot_tooling.h`?

File: .vscode/launch.json (hunk 1 above)
Comment (FrederikTobner) [reply .vscode/launch.json:1]: We should but i want to use my own launch .config and if every dev would do for differrent compilers / os the file will get very long. Maybe we should provide one or two examples in the technical documentation or the README.md instead.

Comment (jksevend) [reply .vscode/launch.json:1]: So let's add it to the .gitignore also right?

**Comment by FrederikTobner** [comment]
> Please also rename `tobot.h` to `tobot_engine.h`. Is it possible to create a separate include directory for `tobot_tooling` where all headers are exported through a `tobot_tooling.h`?

Yes of course

File: tobot_tooling/src/CMakeLists.txt (hunk 4 above)
Comment (FrederikTobner) [reply tobot_tooling/src/CMakeLists.txt:2]: No forgot to add it 🥴

**Review by jksevend — APPROVED** [review]

**Pull request merged**